From fa042fe5c3a65d6d2eee7270d7c5c3cb413ef2ed Mon Sep 17 00:00:00 2001
From: Michael Fladischer <fladi@debian.org>
Date: Thu, 8 Oct 2015 08:37:03 -0700
Subject: use local objects.inv where possible

 Upstream uses intersphinx mappings that fetch the objects.inv for python,
 kombu and django-celery by HTTP from a remote host. Using the local
 objects.inv from python and kombu enables the package to build without
 network connection.
Forwarded: not-needed
Last-Update: 2011-12-27

Patch-Name: intersphinx.patch
---
 docs/conf.py | 24 +++++++++++++++++-------
 1 file changed, 17 insertions(+), 7 deletions(-)

diff --git a/docs/conf.py b/docs/conf.py
index 551e7a6..b2722d8 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -75,13 +75,23 @@ unused_docs = [
 # If true, '()' will be appended to :func: etc. cross-reference text.
 add_function_parentheses = True
 
-intersphinx_mapping = {
-    'python': ('http://docs.python.org/dev', None),
-    'kombu': ('http://kombu.readthedocs.org/en/latest/', None),
-    'djcelery': ('http://django-celery.readthedocs.org/en/latest', None),
-    'cyme': ('http://cyme.readthedocs.org/en/latest', None),
-    'amqp': ('http://amqp.readthedocs.org/en/latest', None),
-}
+def check_object_path(key, url, path):
+    if os.path.isfile(path):
+        return {key: (url, path)}
+    return {}
+
+intersphinx_mapping = {}
+intersphinx_mapping.update(check_object_path('python',
+                                            'http://docs.python.org/',
+                                            '/usr/share/doc/python'
+                                              + '.'.join([str(x) for x in sys.version_info[0:2]])
+                                              + '/html/objects.inv'))
+intersphinx_mapping.update(check_object_path('kombu',
+                                             'http://kombu.readthedocs.org/en/latest/',
+                                             '/usr/share/doc/python-kombu-doc/html/objects.inv'))
+intersphinx_mapping.update(check_object_path('amqp',
+                                             'http://amqp.readthedocs.org/en/latest/',
+                                             '/usr/share/doc/python-amqp-doc/html/objects.inv'))
 
 # The name of the Pygments (syntax highlighting) style to use.
 pygments_style = 'colorful'
